### PR TITLE
Add (and make default) a uuidToken option to use uuidgen to force

### LINKED ISSUE #574: k2 etcd cluster names and tokens may not be unique
Each etcd instance is started with a cluster token set within `ansible/roles/kraken.nodePool/kraken.etcd.docker/templates/etcd.units.etcd.part.jinja2 `:

```
     --discovery-srv {{cluster_node_tuple.1.name}}.{{cluster_node_tuple.0.name}}.internal \
...
     --initial-cluster-token {{cluster_node_tuple.1.etcdConfig.clusterToken}} \
```

Since the clusterToken is fixed, two k2 clusters with the same name could attempt to form an etcd cluster together. Note that this might not affect our CI, because, at least for the k2 PR job, the job number is included in the cluster name (e.g. `sed -i -e s/- name:[[:space:]]*$/- name: master-138/ cluster/aws/config.yaml`).

For more information on etcd discovery, see the [Clustering Guide](https://coreos.com/etcd/docs/latest/v2/clustering.html#dns-discovery) and the [Discovery Service Protocol](https://coreos.com/etcd/docs/latest/v2/discovery_protocol.html).

This is related to https://github.com/samsung-cnct/k2/issues/353 and https://github.com/samsung-cnct/k2/pull/351 (which introduced the problem). It may or may not be fixed by the solution to https://github.com/samsung-cnct/k2/issues/559.

### PR DESCRIPTION
**Pull request by davidewatson**
each cluster formation attempt to occur exactly once. Fixes #574.

### CODE REVIEW

**Comment by davidewatson** [comment]
I wonder if https://github.com/samsung-cnct/k2/pull/685 is necessary with this PR. I don't think it can hurt, but if we use these names for any administrative purpose, they are going to be a pain to verify and type.

Looking at https://coreos.com/etcd/docs/latest/v2/clustering.html (old, I know, but I think still default for our configs),

1) For DNS discovery (SRV) the names must match one of those in the SRV record. Why not just make the names the same as the hostnames (e.g. etcd-1, etcd-2, etc.) so that they never have to change?

2) For public discovery,

> Each member must have a different name flag specified. Hostname or machine-id can be a good choice. Or discovery will fail due to duplicated name.

But we don't use it that public discovery.

I'd like to think further about the ramifications of including IP addresses in the name. It can be useful if the IP address has meaning (e.g. the physical position of a node in a rack), but in the cloud, that's rarely the case.

**Comment by davidewatson** [comment]
@joejulian pointed out that instead of using uuidgen, we could rely on the ansible [password module](http://docs.ansible.com/ansible/latest/playbooks_lookups.html#the-password-lookup) instead. As long as we limit the character set, that seems reasonable to me.

File: schemas/config/v1/etcdConfig.json [hunk 1]
```diff
@@ -25,10 +25,15 @@
       "default": [ 2379, 4001 ]
     },
     "clusterToken": {
-      "description": "Token used during etcd cluster formation",
+      "description": "Token used during etcd cluster formation. This may be unsafe if an etcd cluster formation has to be restarted, or multiple clusters are created within the same VPC. It is highly recommended that the uuidToken option be used instead.",
       "type": "string",
       "default": "espouse-monger-rarely"
     },
+    "uuidToken": {
```
Comment (leahnp) [inline schemas/config/v1/etcdConfig.json]: So if we want to use this option what do we put in the config? If it's a token, where/how do we get/generate it?

Comment (davidewatson) [inline schemas/config/v1/etcdConfig.json]: You are right that the option may be strangely named. `uuidToken` was chosen to parallel the existing `clusterToken`. The semantics are different insofar as `uuidToken` indicates a procedure which will be used to generate the token, and not the token itself.

I have changed the type of the property and updated the schema description to clarify. Does this suffice?

Comment (leahnp) [reply schemas/config/v1/etcdConfig.json]: Yep that makes sense - thanks!

**Pull request merged**